assets/js/pages/ActivityLogPage/ActivityLogPage.jsx

```diff
@@ -1,39 +1,98 @@
 import React, { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
 
 import PageHeader from '@common/PageHeader';
 import ActivityLogOverview from '@common/ActivityLogOverview';
+import ComposedFilter from '@common/ComposedFilter';
 
 import { getActivityLog } from '@lib/api/activityLogs';
+import { ACTIVITY_TYPES_CONFIG } from '@lib/model/activityLog';
+
+const filters = [
+  {
+    key: 'type',
+    type: 'select',
+    title: 'Resource type',
+    options: Object.entries(ACTIVITY_TYPES_CONFIG).map(([key, value]) => [
+      key,
+      value.label,
+    ]),
+  },
+  {
+    key: 'from_date',
+    title: 'From date',
+    type: 'date',
+    prefilled: true,
+  },
+];
+
+const removeUndefinedFields = (obj) =>
+  Object.fromEntries(
+    Object.entries(obj).filter(([_, v]) => typeof v !== 'undefined')
+  );
+
+const searchParamsToFilters = (searchParams) =>
+  [...searchParams.entries()].reduce(
```

Comment:
**TIL** being `searchParams` an instance of `URLSearchParams`, `searchParams.entries()` returns an `Iterator` object. Or better: an object that implements the `Iterator` interface.

When running on the browser (Chrome), `searchParams.entries().reduce` is fine as it returns an `Array` object (which implements `Iterator`), to mimic the behavior of `Object.entries`.

However, when running tests `searchParams.entries().reduce` is not defined, as Node implements [its own `Iterator` class](https://github.com/nodejs/node/blob/123693ccbf97df37f928a8ec3e39b85d5e719a7c/lib/internal/url.js#L217).

Thus the need to convert the iterator into an Array :shrug: